modules/stochastic_tools/src/userobjects/InverseMapping.C

```diff
@@ -110,13 +110,15 @@ InverseMapping::initialSetup()
 void
 InverseMapping::execute()
 {
-  NonlinearSystemBase & nl = _fe_problem.getNonlinearSystemBase();
-
-  // We create a temporary solution vector that will store the reconstructed solution.
-  std::unique_ptr<NumericVector<Number>> temporary_vector = nl.solution().zero_clone();
-
   for (auto var_i : index_range(_var_names_to_reconstruct))
   {
+    MooseVariableFieldBase * var_to_fill = _variable_to_fill[var_i];
+    MooseVariableFieldBase * var_to_reconstruct = _variable_to_reconstruct[var_i];
+
+    // We create a temporary solution vector that will store the reconstructed solution.
+    std::unique_ptr<NumericVector<Number>> temporary_vector =
+        var_to_reconstruct->sys().solution().zero_clone();
```

Comment:
This should probably be outside the variable loop.

Reply:
The variable can be either the an auxvariable or a nonlinear variable, so unfortunately, the system solution vector depends on the variable in this case. I agree, this can be wasteful if we have a huge monolithic system (1000s of variables) with variables that have only a few dofs.

Reply:
Oh wait, I can probably just build a map at initialization so we can go fast at reconstruction.